### ISSUE TITLE
Failed to fetch market cap history

### ISSUE THREAD

**Issue by zjb0807** [blockscout/blockscout#9586]
### Description

<img width="1439" alt="image" src="https://github.com/blockscout/blockscout/assets/16496415/3da5c8d5-4634-45e3-820a-888c31e8b971">

I don’t know what data was returned by the previous API(#8560). Now it seems that the fields returned have changed.

https://api.coingecko.com/api/v3/coins/ethereum/market_chart?vs_currency=usd&days=3

https://github.com/blockscout/blockscout/blob/c81f9004ab6f696f589bf6f1ed5cb4fbbd2934f2/apps/explorer/lib/explorer/exchange_rates/source/coin_gecko.ex#L27-L30

### Type of the installation

Docker-compose

### Type of the JSON RPC archive node

.

### Backend version

v6.2.0

### Elixir & Erlang/OTP versions

./docker/Dockerfile

**Comment by GimluCom** [comment]
It does not look like the API changed
https://github.com/blockscout/blockscout/blob/ee8268787e3d2564f92a4663b33cb481e333c9a6/apps/explorer/lib/explorer/market/history/source/market_cap/coin_gecko.ex#L41-L50

**Comment by zjb0807** [comment]
@GimluCom You are right, I saw it wrong. Is there any way to log http error messages?

**Comment by GimluCom** [comment]
Would you mind sharing your ["exchange rates" ENVs](https://docs.blockscout.com/for-developers/information-and-settings/env-variables#exchange-rates-management)?

**Comment by zjb0807** [comment]
The complete configuration is [here,](https://github.com/AcalaNetwork/blockscout/blob/develop/docker-compose/docker-compose-mandala-testnet.yml#L94-L101) you need to start postgre and rpc node separately.
The free API of COINGECKO is used, which may have limitations, but because there is no error message, it is difficult to debug. I hope the error message can be added. like this:
<img width="995" alt="image" src="https://github.com/blockscout/blockscout/assets/16496415/80b64200-6a42-4c1c-87a4-b348f34ea462">

**Comment by zjb0807** [comment]
```
docker run -it -d \
  --env POSTGRES_DB=blockscout \
  --env POSTGRES_USER=blockscout \
  --env POSTGRES_PASSWORD=blockscout \
  -p 5432:5432 \
  postgres:15 \
  postgres -c 'max_connections=200' -c 'client_connection_check_interval=60000'


docker run --env-file=./envs/common-blockscout.env \
  --env FIRST_BLOCK=30000000 \
  --env ETHEREUM_JSONRPC_HTTP_URL=https://rpc.ankr.com/eth \
  --env DATABASE_URL=postgresql://blockscout:blockscout@192.168.50.62:5432/blockscout \
  --env DISABLE_EXCHANGE_RATES=false \
  --env EXCHANGE_RATES_COINGECKO_PLATFORM_ID=ethereum \
  --env EXCHANGE_RATES_COINGECKO_COIN_ID=ethereum \
  --env TOKEN_EXCHANGE_RATE_INTERVAL=2m \
  --env EXCHANGE_RATES_MARKET_CAP_SOURCE=coin_gecko \
  --env INDEXER_EMPTY_BLOCKS_SANITIZER_BATCH_SIZE=1 \
  --env INDEXER_CATCHUP_BLOCKS_BATCH_SIZE=1 \
  --env INDEXER_CATCHUP_BLOCKS_CONCURRENCY=1 \
  --env INDEXER_CATCHUP_BLOCK_INTERVAL=30s \
  --env INDEXER_BLOCK_REWARD_BATCH_SIZE=1 \
  --env INDEXER_BLOCK_REWARD_CONCURRENCY=1 \
  --env INDEXER_RECEIPTS_BATCH_SIZE=1 \
  --env INDEXER_RECEIPTS_CONCURRENCY=1 \
  --env INDEXER_COIN_BALANCES_BATCH_SIZE=1 \
  --env INDEXER_COIN_BALANCES_CONCURRENCY=1 \
  --env INDEXER_TOKEN_CONCURRENCY=1 \
  --env INDEXER_TOKEN_BALANCES_BATCH_SIZE=1 \
  --env INDEXER_TOKEN_BALANCES_CONCURRENCY=1 \
  --env INDEXER_CATCHUP_MISSING_RANGES_BATCH_SIZE=1 \
  --env MIN_MISSING_BLOCK_NUMBER_BATCH_SIZE=1 \
  --env INDEXER_DISABLE_PENDING_TRANSACTIONS_FETCHER=true \
  --env INDEXER_DISABLE_INTERNAL_TRANSACTIONS_FETCHER=true \
  blockscout/blockscout:6.2.2.commit.f8092ac3 \
  sh -c "bin/blockscout eval \"Elixir.Explorer.ReleaseTasks.create_and_migrate()\" && bin/blockscout start"
```
this can reproduce the error:
<img width="878" alt="image" src="https://github.com/blockscout/blockscout/assets/16496415/1460df1c-0aa8-4255-95b1-a8422897bdb7">

**Comment by GimluCom** [comment]
> this can reproduce the error

The Coingecko API is rate limiting your IP address because you have not provided an API key. The Coingecko API has a monthly limit, meaning you will be rate limited until the next month.

Please clarify if this solves your issue or if your issue is about including the [`error_code`](https://docs.coingecko.com/v3.0.1/reference/common-errors-rate-limit#common-errors)/`error_message` in the warning from Blockscout.

**Comment by zjb0807** [comment]
I hope to print the error code and error message returned by the API in the log, like this https://github.com/blockscout/blockscout/issues/9586#issuecomment-2002383742
so the developers can resolve it easily.

**Comment by vbaranov** [comment]
We added CoinMaketCap, CryptoRank options for fetching market cap history. Please take a look to the exchange rates management section https://docs.blockscout.com/setup-and-run-blockscout/env-variables/backend-env-variables#exchange-rates. I am closing this issue for now. Please let us know, if you still experience issues with fetching market cap.
